### PR TITLE
Update the DG with user stories, glossary, NFRs, use cases

### LINKED ISSUE #6: Update the DG with user stories, glossary, NFRs, use cases

### PR DESCRIPTION
**Pull request by PohSayKeong**
Closes #6

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -308,18 +313,72 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
 3.  A user with above average typing speed for regular English text (i.e. not code, not system admin commands) should be able to accomplish most of the tasks faster using commands than using the mouse.
+4.  Data should persist across user sessions
+5.  Project should be able to handle information from across academic years
 
 *{More to be added}*
 
 ### Glossary
 
+* **Course**: A program students are enrolled in to work towards a degree
+* **Teaching Assistant**: Students who support the teaching of a course
```
Comment (lshaoqin) [inline docs/DeveloperGuide.md]: Maybe we should clarify here that this is what TA stands for.

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -308,18 +313,72 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
 
       Use case resumes at step 2.
 
+**Use case: Update Contact Information**
+
+**MSS**
+
+1.  User requests to list TAs
+2.  AddressBook shows a list of TAs
+3.  User requests to update the contact of a specific TA in the list and key in the necessary information
```
Comment (lshaoqin) [inline docs/DeveloperGuide.md:322]: Can be broken into two steps, so that if information keyed in is invalid, the user doesn't have to select the TA again

**Review by lshaoqin — CHANGES_REQUESTED** [review]
LGTM, a few small suggestions

File: docs/DeveloperGuide.md (hunk 2 above)
Comment (PohSayKeong) [reply docs/DeveloperGuide.md:322]: for this one I am following the current implementation of AB3, breaking into two steps require code changes so we will have to discuss if we want to go ahead with that

File: docs/DeveloperGuide.md (hunk 1 above)
Comment (PohSayKeong) [reply docs/DeveloperGuide.md]: resolved in [c7f1d00](https://github.com/AY2324S1-CS2103T-T10-1/tp/pull/9/commits/c7f1d00e7e2af5f91627cdf126c76fc7c115e335)

**Review by lshaoqin — APPROVED** [review]
Ok, LGTM!

**Pull request merged**